### ISSUE TITLE
BUG and REGRESSION: Custom NavigationViewItem crashes app at startup

### ISSUE THREAD

**Issue by knightmeister** [microsoft/microsoft-ui-xaml#8814]
### Describe the bug

Firstly, I have spent considerable time investigating this and pinpointing the cause of this issue. Please don't arbitrarily close this without investigating it.

Between WASDK 1.1.5 and 1.2.221109.1 there is a regression that crashes my app at startup. My `NavigationView` uses custom `NavigationViewItem`s (called `HomeMenuItem`). Setting the `NavigationView` property `SelectedItem` to this at startup will cause the app to crash with a nonsensical error `Value does not fall within the expected range.` in everybodies favourite useless block of code:

```
#if DEBUG && !DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION
            UnhandledException += (sender, e) =>
            {
                if (global::System.Diagnostics.Debugger.IsAttached) global::System.Diagnostics.Debugger.Break();
            };
#endif
```

Interestingly, if the ctor assignment of `SelectedItem` is commented and it's done later (i.e. via a button), it works. 

To further elucidate on the cause, overriding the `HomeMenuItem` template (i.e. by setting `Content="Hello"`) in the XAML stops this bug.

The fact that this behaviour occurs during startup is also interesting but works once the app is displayed. It's also clearly a regression given it works in 1.1.5 and earlier (code has been in place since before WASDK).

Please reach out if you need anything further :)

### Steps to reproduce the bug

Github repro available here: https://github.com/knightmeister/WinUiNavViewCooked

Change the Nuget WASDK version to 1.1.5 and observe it works. Anything after that (including the latest) does not work.

### Expected behavior

App does not crash when using custom `NavigationViewItem`.

### NuGet package version

WinUI 3 - Windows App SDK 1.4.0: 1.4.230822000

### Windows version

Windows 11 (22H2): Build 22621

**Comment by knightmeister** [comment]
@llongley any idea what release this will make it into?

**Comment by codendone** [comment]
@knightmeister The fix will be in 1.5.
